app/src/main/java/org/metabrainz/android/presentation/features/listens/ListensViewModel.kt

```diff
@@ -0,0 +1,48 @@
+package org.metabrainz.android.presentation.features.listens
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import org.metabrainz.android.data.repository.ListensRepository
+import org.metabrainz.android.data.sources.api.entities.CoverArt
+import org.metabrainz.android.data.sources.api.entities.listens.Listen
+import org.metabrainz.android.util.Resource
+import javax.inject.Inject
+
+@HiltViewModel
+class ListensViewModel @Inject constructor(val repository: ListensRepository) : ViewModel() {
+    var listens: List<Listen> by mutableStateOf(listOf())
+    var coverArt: CoverArt? by mutableStateOf(null)
```

Comment:
I was unable to run the PR currently but this looks wrong. You probably need a list or a map here to maintain the cover arts of all cover arts for listens. I'll try to find an official example on how to do this but in the meanwhile https://stackoverflow.com/questions/66516702/lazycolumn-is-not-keeping-the-state-of-items-when-scrolling looks relevant.

Reply:
Thanks for the review! I have fixed this issue by fetching all the cover arts and maintaining them in a list. All works great!